python/pyarrow/tensor.pxi

```diff
@@ -702,3 +705,156 @@ shape: {0.shape}""".format(self)
     @property
     def non_zero_length(self):
         return self.stp.non_zero_length()
+
+
+cdef class SparseCSFTensor:
+    """
+    A sparse CSF tensor.
+    """
+
+    def __init__(self):
+        raise TypeError("Do not call SparseCSFTensor's constructor directly, "
+                        "use one of the `pyarrow.SparseCSFTensor.from_*` "
+                        "functions instead.")
+
+    cdef void init(self, const shared_ptr[CSparseCSFTensor]& sp_sparse_tensor):
+        self.sp_sparse_tensor = sp_sparse_tensor
+        self.stp = sp_sparse_tensor.get()
+        self.type = pyarrow_wrap_data_type(self.stp.type())
+
+    def __repr__(self):
+        return """<pyarrow.SparseCSFTensor>
+type: {0.type}
+shape: {0.shape}""".format(self)
+
+    @classmethod
+    def from_dense_numpy(cls, obj, dim_names=None):
+        """
+        Convert numpy.ndarray to arrow::SparseCSFTensor
+        """
+        return cls.from_tensor(Tensor.from_numpy(obj, dim_names=dim_names))
+
+    @staticmethod
+    def from_numpy(data, indptr, indices, shape, axis_order=None,
+                   dim_names=None):
+        """
+        Create arrow::SparseCSFTensor from numpy.ndarrays
+        """
+        cdef shared_ptr[CSparseCSFTensor] csparse_tensor
+        cdef vector[int64_t] c_axis_order
+        cdef vector[int64_t] c_shape
+        cdef vector[c_string] c_dim_names
+
+        for x in shape:
+            c_shape.push_back(x)
+        if not axis_order:
+            axis_order = list(range(len(shape)))
```

Comment:
Isn't it better to use a heuristic based on dimension sizes here?

Reply:
Good idea, changing to `argsort` heuristic.
Alternatively we can also just raise as `axis_order` is quite critical. I slightly prefer `argsort`.
I'm thinking to also add `SparseCSFTensor.axis_order` property.

Reply:
Yes, I think `argsort` is more user-friendly.